secretmanager/api-client/create_secret.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+
+# Copyright 2019 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+"""
+command line application and sample code for creating a new secret.
+"""
+
+import argparse
+
+
+# [START secretmanager_create_secret]
+def create_secret(project_id, secret_id):
+    """
+    Create a new secret with the given name. A secret is a logical wrapper
+    around a collection of secret versions. Secret versions hold the actual
+    secret material.
+    """
+
+    # Import the Secret Manager client library.
+    from google.cloud import secretmanager_v1beta1 as secretmanager
+
+    # Create the Secret Manager client.
+    client = secretmanager.SecretManagerServiceClient()
+
+    # Build the resource name of the parent project.
+    parent = client.project_path(project_id)
+
+    # Create the secret.
+    response = client.create_secret(parent, secret_id, {
+        'replication': {
+            'automatic': {},
```

Comment:
Is this supposed to be an empty dict?

Reply:
Yes. The API is a bit weird here because replication is a oneOf automatic or user_managed. user_managed has options where automatic does not.